src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/Utils.scala

```diff
@@ -307,4 +308,30 @@ object Utils {
         }
     } yield ()
   }
+
+  def pollEServiceById(
+    fetchFunc: => Future[CatalogProcess.EService],
+    condition: CatalogProcess.EService => Boolean,
+    maxRetries: Int = 10,
+    delay: FiniteDuration = 1.second
+  )(implicit ec: ExecutionContext): Future[Unit] = {
+
+    def poll(attempt: Int): Future[Unit] = {
+      if (attempt > maxRetries) {
+        Future.failed(new RuntimeException("Max retries reached"))
+      } else {
+        fetchFunc.flatMap { result =>
+          if (condition(result)) {
+            Future.successful(())
+          } else {
+            akka.pattern.after(delay, using = akka.actor.ActorSystem("system").scheduler) {
+              poll(attempt + 1)
+            }
```

Comment:
we need to be careful when using the actor system, this is probably different from the one used by the application.
I know it's terrible, but feel free to use something like this
```suggestion
            Future { Thread.sleep(delay.millis) }.flatMap(_ => poll(attempt + 1))
```